### ISSUE TITLE
Undefined values should not be used as CLI arg values in drivers

### ISSUE THREAD

**Issue by mykola-mokhnach** [appium/appium#16165]
## The problem

1. Install Appium2+xcuitest driver
2. Start appium server without arguments 
3. Start a driver session
4. Check logs

Actual result:
There are log lines about using undefined values as CLI arg values

Expected result:
CLI arg values must only be considered if they are defined

## Environment

* Appium version (or git revision) that exhibits the issue: 2.0-beta23
* Last Appium version that did not exhibit the issue (if applicable):
* Desktop OS/version used to run Appium:
* Node.js version (unless using Appium.app|exe):
* Npm or Yarn package manager:
* Mobile platform/version under test:
* Real device or emulator/simulator:
* Appium CLI or Appium.app|exe:

## Details

```
[HTTP] --> POST /wd/hub/session
[HTTP] {"capabilities":{"firstMatch":[{}],"alwaysMatch":{"appium:app":"/Users/elf/code/java-client/build/resources/test/apps/TestApp.app.zip","appium:automationName":"XCuiTest","appium:commandTimeouts":"240000","appium:deviceName":"iPhone 12","appium:platformVersion":"15.2","appium:wdaLaunchTimeout":240000,"platformName":"ios"}}}
[debug] [W3C] Calling AppiumDriver.createSession() with args: [null,null,{"firstMatch":[{}],"alwaysMatch":{"appium:app":"/Users/elf/code/java-client/build/resources/test/apps/TestApp.app.zip","appium:automationName":"XCuiTest","appium:commandTimeouts":"240000","appium:deviceName":"iPhone 12","appium:platformVersion":"15.2","appium:wdaLaunchTimeout":240000,"platformName":"ios"}}]
[debug] [BaseDriver] Event 'newSessionRequested' logged at 1637910174511 (08:02:54 GMT+0100 (Central European Standard Time))
[Appium] Attempting to find matching driver for automationName 'XCuiTest' and platformName 'ios'
[Appium] The 'xcuitest' driver was installed and matched caps.
[Appium] Will require it at /Users/elf/.appium/appium-xcuitest-driver/node_modules/appium-xcuitest-driver
[Appium] Appium v2.0.0-beta.23 creating new XCUITestDriver (v3.57.0) session
[Appium] Checking BaseDriver versions for Appium and XCUITestDriver
[Appium] Appium's BaseDriver version is 8.2.2
[Appium] Could not determine XCUITestDriver's BaseDriver version
[debug] [BaseDriver] Creating session with W3C capabilities: {
[debug] [BaseDriver]   "alwaysMatch": {
[debug] [BaseDriver]     "platformName": "ios",
[debug] [BaseDriver]     "appium:app": "/Users/elf/code/java-client/build/resources/test/apps/TestApp.app.zip",
[debug] [BaseDriver]     "appium:automationName": "XCuiTest",
[debug] [BaseDriver]     "appium:commandTimeouts": "240000",
[debug] [BaseDriver]     "appium:deviceName": "iPhone 12",
[debug] [BaseDriver]     "appium:platformVersion": "15.2",
[debug] [BaseDriver]     "appium:wdaLaunchTimeout": 240000
[debug] [BaseDriver]   },
[debug] [BaseDriver]   "firstMatch": [
[debug] [BaseDriver]     {}
[debug] [BaseDriver]   ]
[debug] [BaseDriver] }
[BaseDriver] Session created with session id: c0dc45cd-8460-4852-b356-0bd0aed00e93
>>>> [XCUITest] Using webkitDebugProxyPort value of 'undefined' from CLI
>>>> [XCUITest] Using wdaLocalPort value of 'undefined' from CLI
```

**Comment by boneskull** [comment]
@mykola-mokhnach What ports are actually bound here?  Unclear if this is just a logging problem

**Comment by boneskull** [comment]
I think it's not just a logging problem

**Comment by boneskull** [comment]
note that the code which outputs this log message (in xcuitest) should be eliminated.  I think I can open such a PR now.

it would make sense to provide an alternative means for a driver to report the arguments it's getting, perhaps in base-driver.  @jlipps what do you think?  I'm talking about [this](https://github.com/appium/appium-xcuitest-driver/blob/2a59d8592dc070d481c6637cb4480e752f26e6e7/lib/driver.js#L238-L245)

**Comment by jlipps** [comment]
Yeah, dunno if we need any reporting given that the arguments are already shown in the startup info. And just to double check--how are the args coming into the driver now? Are they on `this.opts` ? How can a driver ensure that an arg takes priority over a cap with the same name?

**Comment by boneskull** [comment]
@jlipps I didn't change that; any cli arg comes in via `cliArgs`.

See #16181 for fix, and `xcuitest` will have a forthcoming PR to remove the unneeded old code

**Comment by boneskull** [comment]
resolved via #16181
